lib/env.ts

```diff
@@ -0,0 +1,43 @@
+import { Option, fromNullable } from "fp-ts/lib/Option";
+import * as t from "io-ts";
+
+export const EnvKey = t.union([
+  t.literal("SPID_SESSION_TOKEN"),
+  t.literal("IO_BACKEND_HOST"),
+  t.literal("IO_BACKEND_BASEPATH")
+]);
+
+export type EnvKey = t.TypeOf<typeof EnvKey>;
+
+export const EnvMap = t.record(EnvKey, t.string);
+export type EnvMap = Partial<t.TypeOf<typeof EnvMap>>;
+
+const pick = (obj: any) =>
+  Object.keys(obj).reduce<EnvMap>(
+    (p, k: string) =>
+      EnvKey.decode(k).fold(
+        () => p,
+        (key: EnvKey) => ({ ...p, [key]: obj[key] })
+      ),
+    {}
+  );
+
+const env: EnvMap = { ...pick(process.env) };
+
+export const get = (key: EnvKey): Option<string> => fromNullable(env[key]);
+
+export const set = (key: EnvKey, value: any): void => (env[key] = value);
```

Comment:
setter methods should return a new object instead of mutating the input

```suggestion
export const set = (key: EnvKey, value: any): void => ({ ...env, [key] = value });
```

never mutate arguments

Reply:
The whole point of using this singleton is having a shared store between the setup script and the actual jest test. I can do what you suggest, yet I have to commit such changes somewhere. Suggestions?

Reply:
just export a constant with all the values or use the solution in https://github.com/pagopa/io-api-testing/pull/1#discussion_r387736861 if you need to get them dynamically
(the part that I didn't get is on why jst would fail if the module is not seen as a singleton)

Reply:
so you just 
```ts
import * as env from './lib/env';
console.log(env.HOME);
env.HOME = 'new home';
```
?

_about the singleton part_
nodejs modules are singleton. Maybe it was just me but now it's fine (data is stored into `env` constants inside `lib/env.ts` module, which is allocated for the entire life of the application)

Reply:
why do you need to reassign env ? (generally speaking, if you need to mutate creaet another object)

Reply:
ok, but still this object has to be stored in a _mutable_ location, isn't it?

Reply:
can you please elaborate it a little bit more ? (what do you intend to be stored into a mutable location)

Reply:
From beginning: I have a store, `lib/env.ts`, in which I have **_roughly_** this:
```ts
const env = {};
export const get = (key) => env[key];
export const set = (key, value) => (env[key] = value)
```
The `env` variable is the _mutable location_ i'm referring to. Even if I change `set` into something like
```ts
export const set = (key, value) => ({ ...env, [key]: value });
```
I will then have to
```ts
import { set } from './lib/env.ts';
const env1 = set('foo', 'bar');
// --> where do I save env1?
```

Reply:
what do you mean with "save" ?

Reply:
I'm creating a new version of the "state", I have to store it and make it available to others

Reply:
the conversation continues here: https://github.com/pagopa/io-api-testing/pull/1#discussion_r388394446